### PR TITLE
Add module-info to appconfig

### PR DESCRIPTION
**Pull request by srnagar**
This change is to add module-info.java to AppConfiguration module and update module names to be consistent.

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+module com.azure.data.appconfiguration {
+    requires com.azure.core;
+
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java]: This should `exports com.azure.data.appconfiguration` and `exports com.azure.data.appconfiguration.models`

File: sdk/core/azure-core-http-okhttp/src/main/java/module-info.java [hunk 2]
```diff
@@ -1,16 +1,13 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-module azure.core.http.okhttp {
+module com.azure.core.http.okhttp {
 
     requires transitive com.azure.core;
+    requires com.azure.core.test;
```
Comment (alzimmermsft) [inline sdk/core/azure-core-http-okhttp/src/main/java/module-info.java]: Do we want to make `com.azure.core.test` `requires transitive`?

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java (hunk 1 above)
Comment (srnagar) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java]: Good point! Updated

File: sdk/core/azure-core-http-okhttp/src/main/java/module-info.java (hunk 2 above)
Comment (conniey) [reply sdk/core/azure-core-http-okhttp/src/main/java/module-info.java]: I'm curious why we would want to make it transitive? I thought `transitive` is if you wanted anyone consuming azconfig to implicitly reads azure-core-test. I don't think we ship anything publicly that requires this?

Comment (srnagar) [reply sdk/core/azure-core-http-okhttp/src/main/java/module-info.java]: I don't think anyone wanting to use okhttp module would also want to use test module. If required we can update later.

**Pull request merged**